File: tests/ObjectCompareTest.php
<?php
namespace PHPJava\Tests;

class ObjectCompareTest extends Base
{
    protected $fixtures = [
        'ObjectCompareTest',
    ];

    public function testObjectCompareTest()
    {
        ob_start();
        $calculatedValue = static::$initiatedJavaClasses['ObjectCompareTest']
            ->getInvoker()
            ->getStatic()
            ->getMethods()
            ->call(
                'compareInitiatedObjects'
            );
        $result = trim(ob_get_clean());

        $this->assertEquals(
            'NotSame',
            $result
        );
    }
}

File: tests/fixtures/java/ObjectCompareTest.java
class ObjectCompareTest
{
    public static void compareInitiatedObjects()
    {
        ObjectCompareTest oct1 = new ObjectCompareTest();
        ObjectCompareTest oct2 = new ObjectCompareTest();
        System.out.println(oct1 == oct2 ? "Same" : "NotSame");
    }
}
